packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -299,7 +299,11 @@ export abstract class FunctionBase extends Resource implements IFunction {
               action: 'lambda:InvokeFunction',
             });
 
-            return { statementAdded: true, policyDependable: this._functionNode().findChild(identifier) } as iam.AddToResourcePolicyResult;
+            const permissionNode = this._functionNode().tryFindChild(identifier);
+            if (!permissionNode) {
+              throw new Error('attempted to call `grantInvoke` on unsupported function (either imported or $LATEST version).');
```

Comment:
Nit: something like this maybe, so it's a bit clearer to the user. The user might not know about 'grantInvoke'.

```suggestion
              throw new Error('Cannot modify permission to lambda function. Function is either imported or $LATEST version.');
```